flytekit/clis/sdk_in_container/run.py

```diff
@@ -189,6 +191,8 @@ def _parse_workflow_inputs(click_ctx, wf_entity, create_upload_location_fn: Opti
             value = int(value)
         elif python_type == float:
             value = float(value)
+        elif python_type.__origin__ in {list, dict}:
+            value = json.loads(value)
```

Comment:
in this case, the type engine makes sure this is the correct type when the workflow is executed.

Should there be some up-front type-checking at this point?

Reply:
I guess that is good enough